### PR TITLE
Update post editor trash button

### PR DESCRIPTION
**Pull request by paulwilde**
This addresses a few issues I've noticed:

- The 'Move to Trash' button should be red as it's a destructive action. The hex colour used is referenced in this ([#35622](https://core.trac.wordpress.org/ticket/35622)) core ticket.
- The button text is now correctly centre aligned
- When the 'Move to Trash' button isn't displayed (due to a new post), the padding-top added to each `editor-post-status__row` would leave a blank space, this has been addressed

Empty state before:

screen shot 2017-06-16 at 13 34 56

Empty state after:

screen shot 2017-06-16 at 13 35 18

Button before:

screen shot 2017-06-16 at 13 36 06

Button after:

screen shot 2017-06-16 at 13 35 40

### CODE REVIEW

File: editor/assets/stylesheets/_variables.scss [hunk 1]
```diff
@@ -19,6 +19,7 @@ $light-gray-300: #edeff0;
 $light-gray-200: #f3f4f5;
 $light-gray-100: #f8f9f9;
 $white: #fff;
+$red: #dc3232;
```
Comment (youknowriad) [inline editor/assets/stylesheets/_variables.scss]: There's already an $alert-red color. I wonder if we should use it and update its hex value to match this.

Comment (paulwilde) [inline editor/assets/stylesheets/_variables.scss]: Probably. I'm not sure where the current hex for $alert-red comes from, but #dc3232 is the new standard red in core (Which is also used on the notices).

**Comment by aduth** [comment]
Another option on vertical alignment is Flexbox row with `align-items: center`. Not sure if this is necessarily any better, but it's been used elsewhere and avoids the need for an intermediary span.

```diff
diff --git a/editor/sidebar/post-trash/style.scss b/editor/sidebar/post-trash/style.scss
index 7bf64ae8..4aab704a 100644
--- a/editor/sidebar/post-trash/style.scss
+++ b/editor/sidebar/post-trash/style.scss
@@ -1,11 +1,12 @@
 .editor-post-trash {
+       display: flex;
+       justify-content: flex-end;
+       align-items: center;
        cursor: pointer;
        color: inherit;
        flex-grow: 1;
-       text-align: right;
 
        .dashicon {
-               vertical-align: middle;
                margin-right: -10px;
                margin-left: 10px;
        }
```

File: editor/assets/stylesheets/_variables.scss (hunk 1 above)
Comment (aduth) [inline editor/assets/stylesheets/_variables.scss]: The color was introduced in #924 and may have been ported directly from Calypso. I think it'd be fair to update `$alert-red` to the core standard.

Comment (jasmussen) [inline editor/assets/stylesheets/_variables.scss]: Yes, let's use the $alert-red color, which I'm taking from Hugo's proposed new colors. We are tracking whether to use those, and have them influence the rest of the admin (which I'm a proponent for), or use the others. In the case of the latter it's extra important that it's a variable so it's easy to address if need be. Here are Hugo's colors:

https://codepen.io/hugobaeta/full/RNOzoV/

See also #573

Comment (paulwilde) [inline editor/assets/stylesheets/_variables.scss]: OK, I've gone ahead and switched to the $alert-red colour.

Comment (youknowriad) [inline editor/assets/stylesheets/_variables.scss]: Should we update the alert-colors to match the codepen in this PR?

**Comment by jasmussen** [comment]
Maybe we should loop in @hugobaeta in real quick.

**Comment by paulwilde** [comment]
Is this just waiting for a hex colour to be determined?

**Comment by jasmussen** [comment]
> Is this just waiting for a hex colour to be determined?

My apologies for forgetting about this one. 

Let's go with `$alert-red` for the red color. That way if the red needs to be updated for any reason, we can do it in one place.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Pull request merged**